From 27a73856ecc481c66c7afac8171f753887f32e31 Mon Sep 17 00:00:00 2001
Message-Id: <27a73856ecc481c66c7afac8171f753887f32e31.1342518105.git.minovotn@redhat.com>
From: Pavel Hrdina <phrdina@redhat.com>
Date: Wed, 30 May 2012 19:35:19 +0200
Subject: [PATCH 01/41] remove broken code for tty

RH-Author: Pavel Hrdina <phrdina@redhat.com>
Message-id: <cb99b40d6894c04baa256ee2c3eec2266428f4d7.1338401409.git.phrdina@redhat.com>
Patchwork-id: 39858
O-Subject: [RHEL-6.4 qemu-kvm PATCH 1/4] remove broken code for tty
Bugzilla: 806768
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Luiz Capitulino <lcapitulino@redhat.com>
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>

From: Paolo Bonzini <pbonzini@redhat.com>

This code is taking the settings for a serial port and moving it to
fd 0 when qemu exits.  This is likely just cut-and-paste, rip it.

Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
Signed-off-by: Anthony Liguori <aliguori@us.ibm.com>
(cherry picked from commit d55dbc3acdaf92b42c7fd077c8b6c1392131c043)
Signed-off-by: Pavel Hrdina <phrdina@redhat.com>
---
 qemu-char.c |   10 ----------
 1 files changed, 0 insertions(+), 10 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 qemu-char.c |   10 ----------
 1 file changed, 10 deletions(-)

diff --git a/qemu-char.c b/qemu-char.c
index 91645d2..e88cec4 100644
--- a/qemu-char.c
+++ b/qemu-char.c
@@ -1122,9 +1122,6 @@ static void tty_serial_init(int fd, int speed,
            speed, parity, data_bits, stop_bits);
 #endif
     tcgetattr (fd, &tty);
-    if (!term_atexit_done) {
-        oldtty = tty;
-    }
 
 #define check_speed(val) if (speed <= val) { spd = B##val; break; }
     speed = speed * 10 / 11;
@@ -1296,11 +1293,6 @@ static int tty_serial_ioctl(CharDriverState *chr, int cmd, void *arg)
     return 0;
 }
 
-static void tty_exit(void)
-{
-    tcsetattr(0, TCSANOW, &oldtty);
-}
-
 static void qemu_chr_close_tty(CharDriverState *chr)
 {
     FDCharDriver *s = chr->opaque;
@@ -1335,8 +1327,6 @@ static CharDriverState *qemu_chr_open_tty(QemuOpts *opts)
     }
     chr->chr_ioctl = tty_serial_ioctl;
     chr->chr_close = qemu_chr_close_tty;
-    if (!term_atexit_done++)
-        atexit(tty_exit);
     return chr;
 }
 #else  /* ! __linux__ && ! __sun__ */
-- 
1.7.10.4

